Update LSP0 to latest specification (#492)
lukso-network/docs: docs/standards/universal-profile/lsp0-erc725account.md

- **[LSP17-ContractExtension](../generic-standards/lsp17-contract-extension.md)** enables the contract to be **extended after deployment** to support new standard and functionalities.

> - **[LSP20-CallVerification](#)** provides a unified and standard way for all addresses to **interact directly with the account**, streamlining the interaction process taking into consideration the ownership setup, and enhancing accessibility and developer experience.

Reviewer: Suggesting breaking it down in 2 sentences

```suggestion
- **[LSP20-CallVerification](#)** provides a unified and standard way for all addresses to **interact directly with the account**. This streamlines the interaction process, taking into consideration the ownership setup, and enhancing accessibility and developer experience.
```

However, the account can declare support for new interface IDs after it has been extended. This is especially beneficial for contracts that check if a contract supports a specific interface ID before interacting with it.

![LSP0 Extended interfaceIds](/img/standards/lsp0/LSP0-Extended-Interfaces.jpeg)

### LSP20 - Call Verification

:::tip

See the **[LSP20 - CallVerification](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-20-CallVerification.md)** standard for more information.

Check the **[LSP0 functions](../smart-contracts/lsp0-erc725-account.md)** to see how verification of allowed calls to the account happens.

:::

> The **LSP0ERC725Account** is an ownable contract that can be owned by different types of addresses, including EOAs, contracts like multisigs, KeyManagers, etc.. These owner contracts may have various functions and behaviors, which can create some challenges in terms of figuring out how the interaction with the account works.

Reviewer: ```suggestion
The **LSP0ERC725Account** is an ownable contract that can be owned by different types of addresses, including EOAs, contracts like multisigs, KeyManagers, etc.. These owner contracts may have various functions and behaviours, which can create challenges in figuring out how the interaction with the account works.
```

> Previously, When the account is owned by a KeyManager, for example, only addresses allowed by the KeyManager should interact with it. They cannot directly interact with the account functions because only the KeyManager can call them as the owner. A similar issue arises when the account is owned by a voting contract, as voters must interact with the voting contract rather than the account directly.

Reviewer 2: ```suggestion
When a KeyManager owns the account, only addresses allowed by the KeyManager should interact with it. However, they cannot directly interact with the account functions because only the KeyManager can call them the owner. A similar issue arises when a voting contract owns the account, as voters must interact with the voting contract rather than the account directly.
```
Tried to bring some more clarity, hopefully the meaning is still the same.

> Previously, When the account is owned by a KeyManager, for example, only addresses allowed by the KeyManager should interact with it. They cannot directly interact with the account functions because only the KeyManager can call them as the owner. A similar issue arises when the account is owned by a voting contract, as voters must interact with the voting contract rather than the account directly.

Reviewer: ```suggestion
Previously, When the account was owned by a KeyManager, only addresses allowed by the KeyManager should interact with the account via the Key Manager. They cannot directly interact with the account functions because only the KeyManager can call them as the owner. A similar issue arises when the account is owned by a voting contract, as voters must interact with the voting contract rather than the account directly.
```

> To ensure a unified and standard way to interact with the account, the **LSP20-CallVerification** standard was proposed. This standard aims to streamline the interaction with the account, taking in consideration the ownership setup.

Reviewer 2: ```suggestion
To ensure a unified and standard way to interact with the account, the **LSP20-CallVerification** standard was proposed. This standard aims to streamline the interaction with the account, considering the ownership setup.
```

![LSP0 Old interaction](/img/standards/lsp0/LSP0-OldInteraction.jpeg)

> Currently, When the owner is calling a function on the account contract, they are allowed to directly execute it. However, if a different address calls a function on the LSP0, the function will forward the call to the owner of the account for verification. The owner must then determine if the caller is allowed to execute the specific function.

Reviewer 2: ```suggestion
When the owner calls a function on the account contract, they are allowed to execute it directly. However, if a different address calls a function on the LSP0, the function will forward the call to the account owner for verification. The owner must then determine if the caller can execute the specific function.
```
